src/hal/usbboot.rs

```diff
@@ -79,36 +72,30 @@ pub async fn get_device_path<I: IntoIterator<Item = &'static str>>(
         let Ok(file_name) = entry.file_name().into_string() else {
             continue;
         };
+        let vendor_path = format!("/sys/block/{}/device/vendor", file_name);
+        let Ok(vendor) = tokio::fs::read_to_string(vendor_path).await else {
+            continue;
```

Comment:
i would log an warning here and continue

Reply:
For some devices it's not really a warning, for examplemmcblk device doesn't, and shouldn't, have one.